src/coreclr/dlls/mscoree/exports.cpp

```diff
@@ -189,12 +191,99 @@ static void ConvertConfigPropertiesToUnicode(
             if (hostContractLocal->pinvoke_override != nullptr)
                 *pinvokeOverride = hostContractLocal->pinvoke_override;
         }
+        else if (strcmp(propertyKeys[propertyIndex], HOST_PROPERTY_DIAGNOSTIC_STARTUP_HOOKS) == 0)
+        {
+            diagnostic_startup_hooks_value = propertyValuesW[propertyIndex];
+        }
+    }
+
+    if (nullptr != diagnostic_startup_hooks_value)
+    {
+        HostInformation::SetProperty(HOST_PROPERTY_DIAGNOSTIC_STARTUP_HOOKS, diagnostic_startup_hooks_value);
     }
 
     *propertyKeysWRef = propertyKeysW;
     *propertyValuesWRef = propertyValuesW;
 }
 
+static void CreateAppDomainProperties(
```

Comment:
What I'm trying to do in this function is create a new property list of the app domain creation that composes the original property list passed into `coreclr_intialize` with the new `DIAGNOSTIC_STARTUP_HOOKS` property from the host. My thought is that if I load this property from the property list (if it exists) and set it into the host, allow the diagnostic command to modify it, and then retrieve it again for recomposition, then it'll be the most update to date without having lost any information. I'm not trying to enumerate all properties from the host because that's not what the behavior was before (well, it's the observed behavior because the host is passing the same properties into `coreclr_intialize` but that's not know from this function).

Reply:
Does this need to go through the AppContext property bag at all? It looks unnecessarily complex.

I think it would be a lot simpler for the eventpipe diagnostic command to store the path in a static variable, and the startup hook provider to read it from that static variable.

Reply:
How would I get the startup hook provider to read the static variable? Is the static variable native or managed (I assume it's native and then supplied to `StartupHookProvider` in some way)? Is changing the signature of `StartupHookProvider.MamagedStartup` permitted to take a string parameter? If I'm modifying any managed signatures in `StartupHookProvider`, do I need to make sure Mono works with it?

Reply:
> Is the static variable native or managed (I assume it's native and then supplied to StartupHookProvider in some way)?

Right, it would be native and supplied to `StartupHookProvider`.

> Is changing the signature of StartupHookProvider.ManagedStartup permitted to take a string parameter?

Yes, that's fine. The easiest way to do it is to change it to `void ManagedStartup(char* pDiagnosticStartupHook)` and then create the managed string out of `pDiagnosticStartupHook`.

> I need to make sure Mono works with it?

Yes. (It should not be hard - you can leave it as the last thing to fix.)